x-pack/plugins/apm/public/components/app/alerts_overview/index.tsx

```diff
@@ -6,82 +6,118 @@
-import { environmentQuery } from '../../../../common/utils/environment_query';
-import {
-  AlertsTableStatusFilter,
-  ALL_ALERTS_FILTER,
-  AlertStatusFilterButton,
-} from './alerts_table_status_filter';
+import { getEnvironmentKuery } from '../../../../common/environment_filter_values';
+import { push } from '../../shared/links/url_helpers';
+
+export const ALERT_STATUS_ALL = 'all';
 
 export function AlertsOverview() {
+  const history = useHistory();
   const {
     path: { serviceName },
-    query: { environment },
+    query: { environment, rangeFrom, rangeTo, kuery },
   } = useAnyOfApmParams(
     '/services/{serviceName}/alerts',
     '/mobile-services/{serviceName}/alerts'
   );
   const { services } = useKibana<ApmPluginStartDeps>();
   const [alertStatusFilter, setAlertStatusFilter] =
-    useState<AlertStatusFilterButton>(ALL_ALERTS_FILTER);
+    useState<AlertStatus>(ALERT_STATUS_ALL);
+  const [esQuery, setEsQuery] = useState<{ bool: BoolQuery }>();
 
   const {
     triggersActionsUi: {
       getAlertsStateTable: AlertsStateTable,
+      getAlertsSearchBar: AlertsSearchBar,
       alertsTableConfigurationRegistry,
     },
+    notifications,
+    data: {
+      query: {
+        timefilter: { timefilter: timeFilterService },
+      },
+    },
   } = services;
 
-  const alertQuery = useMemo(
-    () => ({
-      bool: {
-        filter: [
-          {
-            term: { [SERVICE_NAME]: serviceName },
-          },
-          ...(alertStatusFilter !== ALL_ALERTS_FILTER
-            ? [
-                {
-                  term: { [ALERT_STATUS]: alertStatusFilter },
-                },
-              ]
-            : []),
-          ...environmentQuery(environment),
-        ],
-      },
-    }),
-    [serviceName, alertStatusFilter, environment]
-  );
+  const useToasts = () => notifications!.toasts;
+
+  const apmQueries = useMemo(() => {
+    const environmentKuery = getEnvironmentKuery(environment);
+    let query = `${SERVICE_NAME}:${serviceName}`;
 
-  const alertStateProps = {
-    alertsTableConfigurationRegistry,
-    id: 'service-overview-alerts',
-    configurationId: AlertConsumers.OBSERVABILITY,
-    featureIds: [AlertConsumers.APM],
-    query: alertQuery,
-    showExpandToDetails: false,
-  };
+    if (environmentKuery) {
+      query += ` AND ${environmentKuery}`;
+    }
+    return [
+      {
+        query,
+        language: 'kuery',
+      },
+    ];
+  }, [serviceName, environment]);
 
   return (
     <EuiPanel borderRadius="none" hasShadow={false}>
       <EuiFlexGroup direction="column" gutterSize="s">
         <EuiFlexItem>
           <EuiFlexItem grow={false}>
-            <AlertsTableStatusFilter
+            <ObservabilityAlertSearchBar
+              appName={'apmApp'}
+              kuery={kuery}
+              onRangeFromChange={(rangeFrom) =>
+                push(history, {
+                  query: {
+                    rangeFrom,
+                  },
+                })
+              }
```

Comment:
I think this will be better to visualize. WDYT?

```suggestion
                push(history, { query: { rangeFrom, }, }) }
```

Reply:
what do you mean "visualize" ?

Reply:
I meant just to write it in a single line.

Reply:
updated here https://github.com/elastic/kibana/pull/149610/commits/1e899e09de0a3d150c40882f25243fe305700bea